lib/logstash/outputs/loggly.rb

```diff
@@ -117,17 +123,50 @@ def send_event(url, message)
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
     end
 
-    # post message
     request = Net::HTTP::Post.new(url.path)
     request.body = message
-    response = http.request(request)
 
-    if response.is_a?(Net::HTTPSuccess)
-      @logger.info("Event send to Loggly OK!")
-    else
-      @logger.warn("HTTP error", :error => response.error!)
+    # Variable for count total retries
+    totalRetries = 0
+
+    #try posting once when can_retry is false
+    if @can_retry == false
+      @retry_count = 1
+    end
+
+    
+    while totalRetries < @retry_count
+    begin
+
+      response = http.request(request)	
+      case response.code
+	    when "200"					# HTTPSuccess :Code 2xx
+	      puts "Event send to Loggly"
+	    when "403"					# HTTPForbidden :Code 403
+	      @logger.warn("User does not have privileges to execute the action.")
+	    when "404"					# HTTPNotFound :Code 404
+	      @logger.warn("Invalid URL. Please check URL should be http://logs-01.loggly.com/inputs/CUSTOMER_TOKEN/tag/logstash")
+	    when "500"					# HTTPInternalServerError :Code 500
+	      @logger.warn("Internal Server Error")
+	    when "504"					# HTTPGatewayTimeOut :Code 504
+	      @logger.warn("Gateway Time Out")
+	    else
+	      @logger.error("Error! response code is :"+response.code)
+      end # case
+
+    if ["200","403","404"].include?(response.code)	# break the retries loop for the specified response code
+      break
     end
+
+    rescue Exception => e
```

Comment:
You should rescue `StandardError` here, not `Exception`. Exception is for things like KeyboardInterrupt (ctrl+c) and OOM errors, and you don't want to catch those probably :)